### DISCUSSION TITLE
Welcome to embedbuilder Discussions!

### DISCUSSION THREAD

**Discussion by Glitchii** [Glitchii/embedbuilder#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.
  If you want to see what is being worked on or what is on the work list, check the [Projects page](https://github.com/Glitchii/embedbuilder/projects/3)

**Comment by Thibaut12345** [comment]
Hello! I really like the embed builder! I just don't know how to send this message to discord itself. I know it's possible with "request", but otherwise I don't know how to start. can anyone help?

**Reply by Glitchii** [reply]
What I did on my discord bot's website is I added a button to send the embed (which allows choosing a channel). When the button is clicked, it gets the content of the variable 'json' (which currently is set on line 12 at https://github.com/Glitchii/embedbuilder/blob/master/assets/js/index.js#L12 and is updated on change) then sends the content (JSON data) to the backed which loads an embed from that JSON (if you're using discord.js you can just send the object eg. `await channel.send(thObj)` which should send it as an embed, if you are using discord.py you can load from dict first `discord.Embed.from_dict(theDict)` then send). Here's an example of sending to the backed:
```js
fetch('/embed', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ guild: '<guild ID>', json, to: '<chosen channel ID>' })
})
```

**Reply by Thibaut12345** [reply]
I don't quite understand... you can't define discord.js in index.js? do you have discord? so we can talk further there?

**Reply by Glitchii** [reply]
No, you can't have Discord.js in the JS file as that file is a client side file and not a server side file. It is imported by the HTML. I will probably rename it to something other than index.js in the next push to avoid confusion since it's not actually the start file. This is a static website, so you'll have to make your own Index file (leaving the one in assets folder as it is) in the main folder and that will load the HTML, that file is where you put discord.js and it won't be publicly accessible from the inspect tools like the one in assets one.

If you need help, you could look up what a static website is and how to convert it to a non static one.
Since you know how to use Discord.js, I'm assuming you also know how to use NodeJs, Deno, or whatever you are using. You can run your bot the same way but this time also load the website on the front end
Λcє#9258

**Comment by wohahobg** [comment]
Hi,

I think the json format is wrong for embed, it shuld be an array for embeds not single emed as discord api said that https://discord.com/developers/docs/resources/webhook embeds - array of up to 10 embed objects

Will be good if you edit that part of the code, so we can use it direct without editin manual that part. Will be pefect ,

Thanks

**Reply by Glitchii** [reply]
Hello. You should create your own discussion page when making a new discussion.
Anyway, the embed builder did not use an array for embeds because you did not enable multi-embed mode from the menu. Enable it from the menu.

![image](https://user-images.githubusercontent.com/46096865/189739152-acbabe78-86d2-47a1-ba21-f323bed70d8c.png)
